Tests/MongoSwiftSyncTests/MongoCursorTests.swift

```diff
@@ -238,4 +238,40 @@ final class MongoCursorTests: MongoSwiftTestCase {
             expect(cursor.isAlive()).to(beFalse())
         }
     }
+
+    func testCursorTerminatesOnError() throws {
+        try self.withTestNamespace { _, _, coll in
+            try coll.insertOne([:])
+            try coll.insertOne([:])
+
+            let cursor = try coll.find([:], options: FindOptions(batchSize: 1))
+
+            let fp = FailPoint.failCommand(failCommands: ["getMore"], mode: .times(1), errorCode: 10)
+            try fp.enable()
+            defer { fp.disable() }
+
+            var count = 0
+            for result in cursor {
+                expect(count).to(beLessThan(2))
+                if count >= 2 {
```

Comment:
I could not get these tests to halt after failing an `expect`, so I manually added breaks. I tried setting `self.continueAfterFailure = false`, but it didn't seem to work.

Reply:
huh, weird 🤔